### PR TITLE
Add ability to format/add/remove nodes and parameters from python hit

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #10845)

### CODE REVIEW

**Comment by aeslaughter** [comment]
@YaqiWang This should help you format your stuff.

**Comment by YaqiWang** [comment]
Thanks. I will try out this very soon.

File: framework/contrib/hit/parse.cc [hunk 1]
```diff
@@ -183,6 +183,12 @@ Node::~Node()
   }
 }
 
+void
+Node::remove()
+{
+  delete this;
```
Comment (permcody) [inline framework/contrib/hit/parse.cc]: 😆

**Review by permcody — COMMENTED** [review]
My only comment is about the copy in the hit makefile. It seems that if you want a dependency on hit.so in the python directory, the rule to build that file should be elsewhere. Maybe I'm wrong here, but there may be a better way.

File: framework/contrib/hit/parse.cc (hunk 1 above)
Comment (rwcarlsen) [reply framework/contrib/hit/parse.cc]: A quirk of our inability to bridge the python --> c++ bindings gap w.r.t. memory management.

File: framework/contrib/hit/Makefile [hunk 2]
```diff
@@ -34,6 +34,7 @@ bindings: hit.so
 
 hit.so: parse.cc lex.cc braceexpr.cc $(HITCPP)
 	$(CXX) -std=c++11 -w -fPIC -lstdc++ -shared -L$(PYTHONPREFIX)/lib $(PYTHONCFLAGS) $(DYNAMIC_LOOKUP) $^ -o $@
+	cp hit.so ../../../python
```
[the remark this answers is not in the document]
Comment (rwcarlsen) [reply framework/contrib/hit/Makefile]: This makes the hit makefile no longer moose (or location) independent - I think I'd rather we just leave it out.  Normally we don't use this makefile to build hit for moose anyway.

**Comment by permcody** [comment]
Note that this still fails the minimum compiler build.

**Review by rwcarlsen — CHANGES_REQUESTED** [review]
just the makefile thing.

File: framework/contrib/hit/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (aeslaughter) [reply framework/contrib/hit/Makefile]: I did that to same me a step when working, I will remove it.

Comment (aeslaughter) [inline framework/contrib/hit/Makefile]:
```suggestion
```

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**